### ISSUE TITLE
How to prepare the DAIR-V2X data for training?

### ISSUE THREAD

**Issue by linleyzoe** [MediaBrain-SJTU/Where2comm#2]
Hi, thanks for your inspiring work!

There seems no instructions for converting the raw DAIR-V2X to the specific training format, which is referred as  `root_dir: "/GPFS/rhome/quanhaoli/workspace/dataset/my_dair_v2x/v2x_c/cooperative-vehicle-infrastructure/train.json"` in the [config file.](https://github.com/MediaBrain-SJTU/Where2comm/blob/efeee4a33b78c362b277d7f698ad4b3b2da0392c/opencood/hypes_yaml/dair-v2x/dair_where2comm_attn_multiscale_resnet.yaml#L3)

Any advice would be greatly appreciated!

**Comment by goodgamemaker** [comment]
Do you know how to do with it? I also have some questions！

**Comment by suuhou** [comment]
hello, you can find split_data.json files(train.json, val.json) in this zip file [https://github.com/yifanlu0227/CoAlign](url).

Actually, you can also find it on the official github page of dair-v2x. 
[https://github.com/AIR-THU/DAIR-V2X/blob/main/data/split_datas/cooperative-split-data.json](url) 
In this case, you can use the following code to see how they split the dataset into training and validation sets:
```
info_json_path = r"E:\python_work_d\project_samples\DAIR-V2X-main\data\split_datas"
info_name = "cooperative-split-data.json"
with open(os.path.join(info_json_path, info_name), 'rb') as f:
    split_data = json.load(f)
split_info = split_data["cooperative_split"]["train"]
for i in range(10):
    print(split_info[i])
```

Then, to prepare the dataset for model training, following these steps:

1. Download the dair-v2x(cooperative-vehicle-infrastructure )dataset and the new annotation data as mentioned in the github homepage of where2comm
2. Organize your dataset file into the following structure: [https://github.com/AIR-THU/DAIR-V2X/tree/main/data](url)
3. Replace the lidar annotation data of the collaborative and the vehicle side on  with the new annotation data downloaded in step 1. (you will also find the train.json file and the val.json file in this zip )
4. Change data path in the yaml file to your own path (data_dir, root_dir...)

**Comment by CatOneTwo** [comment]
Can you run this program normally?

**Comment by suuhou** [comment]
yes，can't u?

**Comment by CatOneTwo** [comment]
yes, I can run the code now. But how to set communication volume like papers?

**Comment by goodgamemaker** [comment]
> yes, I can run the code now. But how to set communication volume like papers?
Have you solved this problem? Do you use image in the code or only use point cloud？Thank you!

**Comment by goodgamemaker** [comment]
> 
And how does his communication round K reflect

**Comment by W-E-A** [comment]
Google Drive Share Link from CoAlign: https://drive.google.com/file/d/13g3APNeHBVjPcF-nTuUoNOSGyTzdfnUK/view?usp=sharing

**Comment by taipinz** [comment]
D:\DATAZL\DAIR-V2X-C
└─cooperative-vehicle-infrastructure
    ├─cooperative
    │  ├─label_world
    │  └─label_world_backup
    ├─infrastructure-side
    │  ├─calib
    │  │  ├─camera
    │  │  ├─virtuallidar_to_camera
    │  │  └─virtuallidar_to_world
    │  ├─image
    │  ├─label
    │  │  ├─camera
    │  │  └─virtuallidar
    │  └─velodyne
    └─vehicle-side
        ├─calib
        │  ├─camera_intrinsic
        │  ├─lidar_to_camera
        │  ├─lidar_to_novatel
        │  └─novatel_to_world
        ├─image
        ├─label
        │  ├─camera
        │  ├─lidar
        │  └─lidar_backup
        └─velodyne   以上为添加补充注释后DAIR-V2X数据集的文件结构，在训练模型是yaml文件中需要添加train和val的地址，请问该如何划分
